app/controllers/course/material/folders_controller.rb

```diff
@@ -44,8 +44,27 @@ def create_subfolder
     end
   end
 
+  def new_materials
+  end
+
+  def upload_materials
+    if @folder.update_attributes(files_params)
+      redirect_to course_material_folder_path(current_course, @folder),
+                  success: t('.success', name: @folder.name)
+    else
+      flash[:danger] = t('.failure', error: @folder.errors.full_messages.to_sentence)
```

Comment:
this does not make sense. The flash message is shown AFTER the new_materials page is rendered.